22_configuration_management/38_technique_editor.txt

```diff
@@ -0,0 +1,85 @@
+
+===== Utility
+Rudder provides a set of pre-defined Techniques that cover some basic configuration and system administration needs. Of course,this set of techniques cannot responds to all of the specific needs of each client. That’s why Rudder integrate the *Technique editor*, a tool to create advanced Techniques.
+Directly accessible from Ruder menu (_Utilities > Technique editor_), this tool has an easy-to-use interface, which doesn’t require any programming skills but nevertheless allows to create complex Techniques.
+
+===== Interface
+Here is an overview of its interface :
+
+image::./images/technique_editor/1-rudder-technique-editor.png[]
+The interface is divided into 3 columns:
+
+
+.1. A column listing custom Techniques
+image::./images/technique_editor/2-list-techniques.png[]
+Here, we can see our previously created Techniques. We can click on them to see their details/edit them, or create a new one by clicking on the “New” button. Theses Techniques are visible in the *ncf techniques* category in the *Active Techniques tree*, so can be used to create new Directives.
+
+.2. A column with the Technique content
+When we create a new Technique, or when we edit an existing one, the configuration form appears at the center of the interface, instead of the title and the description of the tool.
+image::./images/technique_editor/3-ntp-configuration.png[]
+Then we can see the name, the description, the Bundle name, the version and the Generic methods list of the current Technique. Only the name and the description are editable, the Bundle name and the version are automatically defined during the Technique creation.
+
+.3. A column listing Generic methods / displaying generic method details
+To the right of the interface is the list of Generic methods available for Technique configuration.
+This list is made up of about a hundred Generic methods, grouped according to their category to make them easier to use. (An exhaustive list of them available at any time in the online product documentation can be found on the following link: http://www.rudder-project.org/doc/_generic_methods.html)
+
+image::./images/technique_editor/4-list-generics-method.png[]
+
+You just need to click on a Generic method or drag'n drop it in the area provided for such purpose to add it to the current Technique. Once it's done, you can configure it by clicking on it. Then a new display containing the method details appears instead of the Generic methods list:
+
+image::./images/technique_editor/5-configure-generic-method.png[]
+
+The Generic method details are divided into 3 blocks :
+.Conditions
+Conditions allow user to restrict the execution of the method. 
+.Parameters
+Parameters are in mono or multi line text format. They can contains variables which will be extended at the time of the execution.(syntax:${xxxx})
+.Result classes
+One result class of three will be defined following the execution of a generic method:
+ * One, in case of no modification (Success)
```

Comment:
You could say "Success, when the configuration is correct and no action are needed"